Merge/foundation release/1.13.14+15

meowsbits opened the pull request:
Supersedes #621 

---

## What's Changed

- bump min Go version to 1.21

## Fixed

- 

---

Comparison with last release: [v1.13.1..v1.12.19](https://github.com/etclabscore/core-geth/compare/v1.13.1..v1.12.19)
Docker images published under [etclabscore/core-geth](https://hub.docker.com/r/etclabscore/core-geth/builds).

colinlyguo committed:
ethclient: apply accessList field in toCallArg (#28832)


Co-authored-by: Felix Lange <fjl@twurst.com>

MariusVanDerWijden committed:
params, core/forkid: enable cancun on sepolia and holesky (#28834)

This change enables Cancun 

- Sepolia at 1706655072 (Jan 31st, 2024)
- Holesky at 1707305664 (Feb 7th, 2024)

Specification: https://github.com/ethereum/execution-specs/pull/860

rjl493456442 committed:
core, core/rawdb, eth/sync: no tx indexing during snap sync (#28703)

This change simplifies the logic for indexing transactions and enhances the UX when transaction is not found by returning more information to users.

Transaction indexing is now considered as a part of the initial sync, and `eth.syncing` will thus be `true` if transaction indexing is not yet finished. API consumers can use the syncing status to determine if the node is ready to serve users.

holiman committed:
docs: remove reference to being official (#28858)

gballet committed:
go.{mod,sum}: upgrade go-ole to support arm64 (#28859)

go.{mod,sum}: upgrade go-ole

Commit:
core: fix genesis setup in benchReadChain (#28856)

holiman committed:
all: use uint256 in state (#28598)

This change makes use of uin256 to represent balance in state. It touches primarily upon statedb, stateobject and state processing, trying to avoid changes in transaction pools, core types, rpc and tracers.

sandakersmann committed:
build: upgrade -dlgo version to Go 1.21.6 (#28836)

MariusVanDerWijden committed:
core/state/snapshot: use AddHash/ContainHash instead of Hasher interface (#28849)

This change switches from using the `Hasher` interface to add/query the bloomfilter to implementing it as methods.
This significantly reduces the allocations for Search and Rebloom.

trocher committed:
core/vm: fix misleading comment (#28860)

fix misleading comment

lightclient committed:
eth/catalyst: add timestamp checks to fcu and new payload and improve param checks (#28230)

 This PR introduces a few changes with respect to payload verification in fcu and new payload requests:

* First of all, it undoes the `verifyPayloadAttributes(..)` simplification I attempted in #27872. 
* Adds timestamp validation to fcu payload attributes [as required](https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#specification-1) (section 2) by the Engine API spec. 
* For the new payload methods, I also update the verification of the executable data. For `newPayloadV2`, it does not currently ensure that cancun values are `nil`. Which could make it possible to submit cancun payloads through it. 
* On `newPayloadV3` the same types of checks are added. All shanghai and cancun related fields in the executable data must be non-nil, with the addition that the timestamp is _only_ with cancun.
* Finally it updates a newly failing catalyst test to call the correct fcu and new payload methods depending on the fork.

karalabe committed:
core/txpool, eth/catalyst: fix racy simulator due to txpool background reset (#28837)

This PR fixes an issues in the new simulated backend. The root cause is the fact that the transaction pool has an internal reset operation that runs on a background thread.

When a new transaction is added to the pool via the RPC, the transaction is added to a non-executable queue and will be moved to its final location on a background thread. If the machine is overloaded (or simply due to timing issues), it can happen that the simulated backend will try to produce the next block, whilst the pool has not yet marked the newly added transaction executable. This will cause the block to not contain the transaction. This is an issue because we want determinism from the simulator: add a tx, mine a block. It should be in there.

The PR fixes it by adding a Sync function to the txpool, which waits for the current reset operation (if any) to finish, and then runs an entire round of reset on top. The new round is needed because resets are only triggered by new head events, so newly added transactions will not trigger the outer resets that we can wait on. The transaction pool would eventually internally do a reset even on transaction addition, but there's no easy way to wait on that and there's no meaningful reason to bubble that across everything. A clean outer reset will at worse be a small noop goroutine.

rjl493456442 committed:
core: move tx indexer to its own file (#28857)

This change moves all the transaction indexing functions to a separate txindexer.go file and defines a txIndexer structure as a refactoring.

lightclient committed:
eth/catalyst: prefix payload id with version (#28246)

GetPayloadVX should only return payloads which match its version. GetPayloadV2 is a special snowflake that supports v1 and v2 payloads. This change uses a a version-specific prefix within in the payload id, basically a namespace for the version number.

rjl493456442 committed:
ethclient: fix flaky test (#28864)

Fix flaky test due to incomplete transaction indexing

holiman committed:
params: go-ethereum v1.13.11 stable

holiman committed:
all: release go-ethereum v1.13.11 (#28868)

Release 1.13.11

holiman committed:
params: begin v.1.13.12 release cycle

bodhi-crypo committed:
internal/flags: fix typo (#28876)

protolambda committed:
core/types: fix and test handling of faulty nil-returning signer (#28879)

This adds an error if the signer returns a nil value for one of the signature value fields.

keienWang committed:
README.md: fix travis badge (#28889)

The hyperlink in the README file that directs to the Travis CI build was broken.
This commit updates the link to point to the corrent build page.

lightclient committed:
eth/catalyst: allow payload attributes v1 in fcu v2 (#28882)

At some point, `ForkchoiceUpdatedV2` stopped working for `PayloadAttributesV1` while `paris` was active. This was causing a few failures in hive. This PR fixes that, and also adds a gate in `ForkchoiceUpdatedV1` to disallow `PayloadAttributesV3`.

keienWang committed:
docs/postmortems: fix outdated link (#28893)

rjl493456442 committed:
core: reset tx lookup cache if necessary (#28865)

This pull request resets the txlookup cache if chain reorg happens, 
preventing them from remaining reachable. It addresses failures in
the hive tests.

holiman committed:
build: fix problem with windows line-endings in CI download (#28900)

fixes #28890

rjl493456442 committed:
eth/downloader: fix skeleton cleanup (#28581)

* eth/downloader: fix skeleton cleanup

* eth/downloader: short circuit if nothing to delete

* eth/downloader: polish the logic in cleanup

* eth/downloader: address comments

holiman committed:
deps: update memsize (#28916)

karalabe committed:
core/txpool/blobpool: post-crash cleanup and addition/removal metrics (#28914)

* core/txpool/blobpool: clean up resurrected junk after a crash

* core/txpool/blobpool: track transaction insertions and rejections

* core/txpool/blobpool: linnnnnnnt

karalabe committed:
core/txpool: don't inject lazy resolved transactions into the container (#28917)

* core/txpool: don't inject lazy resolved transactions into the container

* core/txpool: minor typo fixes

zoereco committed:
core/types: fix typo (#28922)

ziogaschr committed:
p2p: fix accidental termination of portMappingLoop (#28911)

karalabe committed:
internal/flags: fix --miner.gasprice default listing (#28932)

rex4539 committed:
all: fix typos in comments (#28881)

Halimao committed:
Makefile: add help target to display available targets (#28845)


Co-authored-by: Martin HS <martin@swende.se>
Co-authored-by: Felix Lange <fjl@twurst.com>

rjl493456442 committed:
core: cache transaction indexing tail in memory (#28908)

karalabe committed:
eth, miner: fix enforcing the minimum miner tip (#28933)

* eth, miner: fix enforcing the minimum miner tip

* ethclient/simulated: fix failing test due the min tip change

* accounts/abi/bind: fix simulater gas tip issue

lmittmann committed:
core/state, core/vm: minor uint256 related perf improvements (#28944)

lightclient committed:
cmd,internal/era: implement `export-history` subcommand (#26621)

* all: implement era format, add history importer/export

* internal/era/e2store: refactor e2store to provide ReadAt interface

* internal/era/e2store: export HeaderSize

* internal/era: refactor era to use ReadAt interface

* internal/era: elevate anonymous func to named

* cmd/utils: don't store entire era file in-memory during import / export

* internal/era: better abstraction between era and e2store

* cmd/era: properly close era files

* cmd/era: don't let defers stack

* cmd/geth: add description for import-history

* cmd/utils: better bytes buffer

* internal/era: error if accumulator has more records than max allowed

* internal/era: better doc comment

* internal/era/e2store: rm superfluous reader, rm superfluous testcases, add fuzzer

* internal/era: avoid some repetition

* internal/era: simplify clauses

* internal/era: unexport things

* internal/era,cmd/utils,cmd/era: change to iterator interface for reading era entries

* cmd/utils: better defer handling in history test

* internal/era,cmd: add number method to era iterator to get the current block number

* internal/era/e2store: avoid double allocation during write

* internal/era,cmd/utils: fix lint issues

* internal/era: add ReaderAt func so entry value can be read lazily

Co-authored-by: lightclient <lightclient@protonmail.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

* internal/era: improve iterator interface

* internal/era: fix rlp decode of header and correctly read total difficulty

* cmd/era: fix rebase errors

* cmd/era: clearer comments

* cmd,internal: fix comment typos

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

lightclient committed:
core,params: add holesky to default genesis function (#28903)

fjl committed:
node, rpc: add configurable HTTP request limit (#28948)

Adds a configurable HTTP request limit, and bumps the engine default

zoereco committed:
all: fix docstring names (#28923)

* fix wrong comment

* reviewers input

* Update log/handler_glog.go

---------

Co-authored-by: Martin HS <martin@swende.se>

bodhi-crypo committed:
ethclient/simulated: fix typo (#28952)

(ethclient/simulated):fix typo

fjl committed:
eth/gasprice: fix percentile validation in eth_feeHistory (#28954)

karalabe committed:
cmd/devp2p, eth: drop support for eth/67 (#28956)

lightclient committed:
params, core/forkid: add mainnet timestamp for Cancun (#28958)

* params: add cancun timestamp for mainnet

* core/forkid: add test for mainnet cancun forkid

* core/forkid: update todo tests for cancun

s1na committed:
internal/ethapi: add support for blobs in eth_fillTransaction (#28839)

This change adds support for blob-transaction in certain API-endpoints, e.g. eth_fillTransaction. A follow-up PR will add support for signing such transactions.

lightclient committed:
internal/era: update block index format to be based on record offset (#28959)

As mentioned in #26621, the block index format for era1 is not in line with the regular era block index. This change modifies the index so all relative offsets are based against the beginning of the block index record.

holiman committed:
params: go-ethereum v1.13.12 stable

holiman committed:
all: release go-ethereum v1.13.12 (#28961)

all: release go-ethereum v1.13.12

holiman committed:
params: begin v1.13.13 release cycle

holiman committed:
build: remove ubuntu 'lunar' build (#28962)

krauspt committed:
fix: update outdated link to trezor docs (#28966)

fix: update link to trezor

mask-pp committed:
internal/ethapi: fix gas estimation bug in eth_fillTransaction for blob tx (#28929)

weiihann committed:
core/txpool/legacypool: use uint256.Int instead of big.Int (#28606)

This change makes the legacy transaction pool use of `uint256.Int` instead of `big.Int`. The changes are made primarily only on the internal functions of legacypool. 

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

lindlof committed:
internal/ethapi, signer/core: fix documentation-links (#28979)

fix: management api links

rjl493456442 committed:
all: remove the dependency from trie to triedb (#28824)

This change removes the dependency from trie package to triedb package.

holiman committed:
cmd/utils: fix merge-breakage in test (#28985)

holiman committed:
tests: fix goroutine leak related to state snapshot generation (#28974)


---------

Co-authored-by: Felix Lange <fjl@twurst.com>

mask-pp committed:
ethereum, ethclient: add blob transaction fields in CallMsg (#28989)


Co-authored-by: Felix Lange <fjl@twurst.com>

bkellerman committed:
core/txpool/blobpool: rename variables in comments (#28981)


Co-authored-by: Felix Lange <fjl@twurst.com>

colinlyguo committed:
cmd/devp2p: fix modulo in makeBlobTxs (#28970)

lightclient committed:
eth/catalyst,beacon/engine: implement GetClientVersionV1 (#28915)

holiman committed:
tests: update execution spec tests + split statetest exec (#28993)

MariusVanDerWijden committed:
eth/catalyst: add getClientVersion to capabilities (#28994)

bodhi-crypo committed:
cmd/evm: fix typo in test script (#28995)

fjl committed:
cmd/devp2p/internal/ethtest: some fixes for the eth test suite (#28996)

Improving two things here:

On hive, where we look at these tests, the Go code comment above the test
is not visible. When there is a failure, it's not obvious what the test is actually
expecting. I have converted the comments in to printed log messages to
explain the test more.

Second, I noticed that besu is failing some tests because it happens to request
a header when we want it to send transactions. Trying the minimal fix here to
serve the headers.

Co-authored-by: lightclient <14004106+lightclient@users.noreply.github.com>

colinlyguo committed:
core/txpool/legacypool: remove a redundant heap.Init (#28910)


Co-authored-by: Martin HS <martin@swende.se>
Co-authored-by: Felix Lange <fjl@twurst.com>

karalabe committed:
core/txpool/blobpool: update the blob db with corruption handling (#29001)

Updates billy to a more recent version which is more robust in the face of corrupt data (e.g. after a hard crash)

s1na committed:
core: move genesis alloc types to core/types (#29003)

We want to use these types in public user-facing APIs, so they shouldn't be in core.

Co-authored-by: Felix Lange <fjl@twurst.com>

karalabe committed:
core/txpool, eth, miner: pre-filter dynamic fees during pending tx retrieval (#29005)

* core/txpool, eth, miner: pre-filter dynamic fees during pending tx retrieval

* miner: fix typo

* core/txpool: handle init-error in blobpool without panicing

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

rjl493456442 committed:
ethstats: prevent panic if head block is not available (#29020)

This pull request fixes a flaw in ethstats which can lead to node crash

A panic could happens when the local blockchain is reorging which causes the original head block not to be  reachable (since number->hash canonical mapping is deleted). In order to prevent the panic, the block nilness is now checked in ethstats.

cuiweixie committed:
core: using math.MaxUint64 instead of 0xffffffffffffffff (#29022)

karalabe committed:
core/txpool, miner: speed up blob pool pending retrievals (#29008)

* core/txpool, miner: speed up blob pool pending retrievals

* miner: fix test merge issue

* eth: same same

* core/txpool/blobpool: speed up blobtx creation in benchmark a bit

* core/txpool/blobpool: fix linter

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

holiman committed:
core/vm, params: ensure order of forks, prevent overflow (#29023)

This PR fixes an overflow which can could happen if inconsistent blockchain rules were configured. Additionally, it tries to prevent such inconsistencies from occurring by making sure that merge cannot be enabled unless previous fork(s) are also enabled.

karalabe committed:
core/txpool, eth, miner: retrieve plain and blob txs separately (#29026)

* core/txpool, eth, miner: retrieve plain and blob txs separately

* core/txpool: fix typo, no farming

* miner: farm all the typos

Co-authored-by: Martin HS <martin@swende.se>

---------

Co-authored-by: Martin HS <martin@swende.se>

buddh0 committed:
core/txpool: fix typo (#29031)

buddh0 committed:
core,eth,internal: fix typo (#29024)

tmelhao committed:
params: add cancun upgrade banner (#29042)

params: add cancun banner

Signed-off-by: tmelhao <tmelhao@gmail.com>
Co-authored-by: tmelhao <tmelhao@gmail.com>

colinlyguo committed:
core/txpool: fix typo (#29036)

* fix typos

* address comments

s1na committed:
internal/ethapi: fix defaults for blob fields (#29037)


Co-authored-by: Martin HS <martin@swende.se>

fjl committed:
params: release go-ethereum v1.13.13 stable

fjl committed:
Merge branch 'master' into release/1.13

fjl committed:
params: begin v1.13.14 release cycle

meowsbits committed:
Merge branch 'release/1.13' into merge/foundation-release/1.13.13

 Conflicts:
	.travis.yml
	Makefile
	README.md
	accounts/abi/bind/backends/simulated.go
	accounts/abi/bind/backends/simulated_test.go
	accounts/abi/bind/bind_test.go
	accounts/abi/bind/util_test.go
	accounts/hd_test.go
	build/ci.go
	cmd/devp2p/internal/ethtest/chain.go
	cmd/devp2p/internal/ethtest/helpers.go
	cmd/devp2p/internal/ethtest/suite_test.go
	cmd/devp2p/internal/ethtest/testdata/genesis.json
	cmd/devp2p/internal/ethtest/transaction.go
	cmd/evm/internal/t8ntool/execution.go
	cmd/evm/internal/t8ntool/transaction.go
	cmd/evm/internal/t8ntool/transition.go
	cmd/evm/main.go
	cmd/evm/runner.go
	cmd/evm/staterunner.go
	cmd/faucet/faucet.go
	cmd/faucet/faucet_test.go
	cmd/geth/chaincmd.go
	cmd/geth/main.go
	cmd/utils/flags.go
	consensus/beacon/consensus.go
	consensus/clique/clique.go
	consensus/clique/clique_test.go
	consensus/ethash/consensus.go
	console/console_test.go
	core/bench_test.go
	core/block_validator_test.go
	core/blockchain_sethead_test.go
	core/blockchain_test.go
	core/chain_makers.go
	core/chain_makers_test.go
	core/forkid/forkid_test.go
	core/gen_genesis_account.go
	core/genesis.go
	core/genesis_test.go
	core/headerchain_test.go
	core/rlp_test.go
	core/state/statedb.go
	core/state_processor.go
	core/state_processor_test.go
	core/state_transition.go
	core/txpool/blobpool/blobpool_test.go
	core/txpool/legacypool/legacypool.go
	core/txpool/legacypool/legacypool_test.go
	core/txpool/validation.go
	core/types/gen_account.go
	core/types/tx_blob.go
	core/vm/evm.go
	core/vm/instructions.go
	core/vm/interface.go
	core/vm/jump_table.go
	core/vm/operations_acl.go
	core/vm/runtime/runtime.go
	docs/postmortems/2021-08-22-split-postmortem.md
	eth/api_debug.go
	eth/api_debug_test.go
	eth/catalyst/api.go
	eth/catalyst/api_test.go
	eth/catalyst/simulated_beacon.go
	eth/catalyst/simulated_beacon_test.go
	eth/downloader/downloader.go
	eth/downloader/downloader_test.go
	eth/downloader/queue.go
	eth/downloader/testchain_test.go
	eth/fetcher/block_fetcher_test.go
	eth/filters/filter_system_test.go
	eth/filters/filter_test.go
	eth/gasprice/gasprice_test.go
	eth/handler.go
	eth/handler_test.go
	eth/protocols/eth/handler.go
	eth/protocols/eth/handler_test.go
	eth/protocols/eth/protocol.go
	eth/protocols/snap/handler_fuzzing_test.go
	eth/tracers/api.go
	eth/tracers/api_test.go
	eth/tracers/internal/tracetest/calltrace_test.go
	eth/tracers/js/tracer_test.go
	eth/tracers/tracers_test.go
	ethclient/ethclient_test.go
	ethclient/gethclient/gethclient_test.go
	go.mod
	go.sum
	graphql/graphql_test.go
	internal/ethapi/api.go
	internal/ethapi/api_test.go
	internal/ethapi/transaction_args.go
	internal/ethapi/transaction_args_test.go
	les/api.go
	les/api_backend.go
	les/api_test.go
	les/benchmark.go
	les/client.go
	les/client_handler.go
	les/commons.go
	les/handler_test.go
	les/odr_test.go
	les/peer.go
	les/peer_test.go
	les/request_test.go
	les/retrieve.go
	les/server.go
	les/state_accessor.go
	les/test_helper.go
	light/lightchain.go
	light/lightchain_test.go
	light/odr.go
	light/odr_test.go
	light/odr_util.go
	light/postprocess.go
	light/trie_test.go
	light/txpool.go
	light/txpool_test.go
	miner/miner_test.go
	miner/stress/clique/main.go
	miner/worker_test.go
	params/config.go
	params/mutations/dao.go
	params/types/genesisT/gen_genesis.go
	params/types/genesisT/gen_genesis_account.go
	params/version.go
	rpc/json.go
	rpc/service.go
	tests/block_test_util.go
	tests/fuzzers/les/les-fuzzer.go
	tests/gen_stenv.go
	tests/init_test.go
	tests/state_test.go
	tests/state_test_util.go

meowsbits committed:
params: unused import; undefined type ChainConfig

Date: 2024-02-21 10:51:34-07:00
Signed-off-by: meows <b5c6@protonmail.com>

ArtificialPB committed:
internal/ethapi: use overriden baseFee for gasPrice (#29051)

eth_call and debug_traceCall allow users to override various block fields, among them base fee. However the overriden base fee was not considered for computing the effective gas price of that message, and instead base fee of the base block was used. This has been fixed in this commit.

colinlyguo committed:
internal/ethapi:fix zero rpc gas cap in eth_createAccessList (#28846)

This PR enhances eth_createAccessList RPC call to support scenarios where the node is launched with an unlimited gas cap (--rpc.gascap 0). The eth_createAccessList RPC call returns failure if user doesn't explicitly set a gas limit.

meowsbits committed:
params/types/genesisT: go run github.com/fjl/gencodec -type GenesisAccount -field-override genesisAccountMarshaling -out gen_genesis_account.go

Date: 2024-02-22 12:21:02-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/gasestimator: undefined: params.ChainConfig (typecheck)

Date: 2024-02-22 12:27:17-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core,core/types,eth/gasestimator,ethclient/simulated: undefined: params.TxGas (typecheck)

Date: 2024-02-22 12:28:16-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/gasestimator: undefined: params.CallStipend (typecheck)

Date: 2024-02-22 12:28:54-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
accounts/abi/bind/backends,cmd/devp2p/internal/ethtest,cmd/faucet,tests/fuzzers/les: expected 'package', found 'EOF' (typecheck)

Date: 2024-02-22 12:30:16-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: `Light*Flag` redeclared in this block (typecheck)

Date: 2024-02-22 12:31:16-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
ethclient/simulated: fix types.Genesis -> genesisT.Genesis typing

Date: 2024-02-22 12:35:51-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: undefined: params.InitialBaseFee (typecheck)

Date: 2024-02-22 12:36:28-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
bind_test,core/types,eth/catalyst,ethclient/simulated,params/vars: undefined: params.GWei (typecheck)

Date: 2024-02-22 12:37:12-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/checkpoint-admin: log.Root().SetHandler undefined

Date: 2024-02-22 12:51:09-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/devp2p/internal/ethtest: gen.ToBlock undefined; + config field fixup

Date: 2024-02-22 12:54:55-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/devp2p/internal/ethtest: ChainConfigurator has no type ChainID

Date: 2024-02-22 12:56:43-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool/blobpool,internal/ethapi,miner,tests: undefined: params.BlobTxBlobGasPerBlob

Date: 2024-02-22 12:58:30-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: undefined: Genesis

Date: 2024-02-22 13:00:13-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: rename trie. -> triedb.

Date: 2024-02-22 13:02:25-07:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
core/types: core/types/gen_account.go:1:1: expected 'package', found 'EOF'

ziogaschr committed:
tests: run go generate

ziogaschr committed:
params/mutations: cannot use uncleRewards[i] (variable of type *big.Int) as *uint256.Int value in argument to state.AddBalance

ziogaschr committed:
core/vm: cannot use host.env.StateDB.GetBalance(common.Address(addr)) (value of type *uint256.Int) as *big.Int value in argument to common.BigToHash

ziogaschr committed:
core: core: cannot use account.Balance (variable of type *big.Int) as *uint256.Int value in argument to statedb.AddBalance

tmelhao committed:
eth/catalyst: fix wrong error message of payloadV2 after cancun (#29049)

* eth/catalyst: the same error format

Signed-off-by: tmelhao <tmelhao@gmail.com>

* eth/catalyst: wrong error message for payloadV2 post-cancun

Signed-off-by: tmelhao <tmelhao@gmail.com>

* eth/catalyst: parentBeaconBlockRoot -> parentBlockBeaconRoot

Signed-off-by: tmelhao <tmelhao@gmail.com>

* apply commit review

Signed-off-by: tmelhao <tmelhao@gmail.com>

---------

Signed-off-by: tmelhao <tmelhao@gmail.com>
Co-authored-by: tmelhao <tmelhao@gmail.com>

ziogaschr committed:
core: undefined: trie.Database

ziogaschr committed:
params/types/genesisT: core,params/types/coregeth,params/types/ctypes,params/types/genesisT,params/types/goethereum: add Verkle transition in configuration

ziogaschr committed:
core: fixes on genesis_test.go

ziogaschr committed:
internal/ethapi: internal/ethapi/transaction_args.go:41:34: undefined: params.MaxBlobGasPerBlock

ziogaschr committed:
core/txpool/legacypool: "github.com/ethereum/go-ethereum/params" imported and not used

ziogaschr committed:
consensus/lyra2: cannot use uncleReward (variable of type *big.Int) as *uint256.Int value in argument to state.AddBalance

ziogaschr committed:
eth/tracers/native: cannot convert t.initialState.GetBalance(addr) (value of type *uint256.Int) to type *hexutil.Big

ziogaschr committed:
miner: ndefined: params

ziogaschr committed:
eth: overrides.OverrideShanghai undefined (type core.ChainOverrides has no field or method OverrideShanghai)

meowsbits committed:
tests: undefined: params.BlobTxGasPerBlob

Date: 2024-02-23 07:06:29-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/devp2p/internal/ethtest: s.chain.config.ChainID undefined

Date: 2024-02-23 07:07:35-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/geth,console,params: fixup DeveloperGenesisBlock and her uses

Date: 2024-02-23 07:12:33-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
les,light: rm -rf

These have been removed by ethereum/go-ethereum
and unless they're necessary for POW
we should remove them too

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
ethclient/simulated: undefined: vars

Date: 2024-02-23 07:22:59-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
accounts/abi/bind/backends,cmd/utils,ethclient/simulated,internal/ethapi: types.GenesisAlloc -> genesisT.GenesisAlloc

Date: 2024-02-23 07:24:02-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core,eth/filters,eth,ethclient,ethclient/gethclient,miner: undefined: trie.NewDatabase (-> triedb.NewDatabase)

Date: 2024-02-23 07:25:26-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/tracers,internal/ethapi: undefined: params.Ether

Date: 2024-02-23 07:26:26-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/downloader: undefined: ETH66

Date: 2024-02-23 07:27:33-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/types,eth/catalyst: undefined: vars

Date: 2024-02-23 07:28:10-07:00
Signed-off-by: meows <b5c6@protonmail.com>

Commit:
core/txpool: reject blob txs with blob fee cap below the minimum (#29081)

* make blobpool reject blob transactions with fee below the minimum

* core/txpool: some minot nitpick polishes and unified error formats

* core/txpool: do less big.Int constructions with the min blob cap

---------

Co-authored-by: Péter Szilágyi <peterke@gmail.com>

0x2d3c committed:
p2p, log, rpc: use errors.New to replace fmt.Errorf with no parameters (#29074)

cuiweixie committed:
eth/catalyst: enable some commented-out testcases   (#29073)

mask-pp committed:
internal/ethapi: pass blob hashes to gas estimation (#29085)

justindhillon committed:
cmd/clef: add spaces in README.md table (#29077)

Add space after links in so they are clickable in vscode.

cuiweixie committed:
eth/catalyst: remove variable in tx conversion loop (#29076)

karalabe committed:
core/txpool/blobpool: reduce default database cap for rollout (#29090)

xcore/txpool/blobpool: reduce default database cap for rollout

jsvisa committed:
core/txpool: no need to log loud rotate if no local txs (#29083)

* core/txpool: no need to run rotate if no local txs

Signed-off-by: jsvisa <delweng@gmail.com>

* Revert "core/txpool: no need to run rotate if no local txs"

This reverts commit 17fab173883168c586d57ca9c05dfcbd9e7831b4.

Signed-off-by: jsvisa <delweng@gmail.com>

* use Debug if todo is empty

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

andreisilviudragnea committed:
eth/tracers: Fix callTracer logs on onlyTopCall == true (#29068)

Commit:
core/txpool: elevate the 'already reserved' error into a constant (#29095)

declare the 'already reserved' error in errors.go

karalabe committed:
params: release Geth v1.13.14

karalabe committed:
Merge branch 'master' into release/1.13

meowsbits committed:
params,params/mutations,params/types/ctypes,params/vars,tests: wip: uint256 re: rewards, state tests

Date: 2024-02-27 07:56:59-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/mutations: finish rewards re: uint256 vs. big

Use uint256 for REWARD values;
use/maintain big for non-REWARD values,
eg. Era and block numbers.

Date: 2024-02-27 08:15:25-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
common/math: implement uint256 common math logic for use w/ core-geth block reward schedules

Date: 2024-02-27 08:35:20-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/coregeth,params/types/ctypes,params/types/genesisT,params/types/goethereum: continue fixing up uint256 re: rewards/schedules

Date: 2024-02-27 08:46:20-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: fixup consensus difficulty calc; difficulty still IS big.Int

But the configurator will ENCODE it as a uint256.
This line (between implementation/encoding::big.Int/uint256.Int)
lets us maintain the Heade's Difficulty type a big.Int,
which it still is upstream, and reuse the reward
en/decoding types.

Date: 2024-02-27 09:00:11-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: undefined: Genesis

Date: 2024-02-27 09:03:21-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: undefined: triedb

Date: 2024-02-27 09:03:40-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: assignment mismatch: 5 variables but t.RunNoVerify returns 3 values

Date: 2024-02-27 09:04:45-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: cannot use t.json.Pre (variable of type stPre) as genesisT.GenesisAlloc value in argument to MakePreState

Date: 2024-02-27 09:05:19-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/tracers: cannot convert minerReward (variable of type *uint256.Int) to type *hexutil.Big

Date: 2024-02-27 09:06:10-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner/stress/ethash: log.Root().SetHandler undefined (type github.com/ethereum/go-ethereum/log.Logger has no field or method SetHandler)

Date: 2024-02-27 09:08:58-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: fixup language re: state.StateDB.Logs

Date: 2024-02-27 09:09:48-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: bc.Config().ChainID undefined (type ctypes.ChainConfigurator has no field or method ChainID)

Date: 2024-02-27 09:10:18-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: genesis.Config.TerminalTotalDifficultyPassed undefined (type ctypes.ChainConfigurator has no field or method TerminalTotalDifficultyPassed)

Date: 2024-02-27 09:11:18-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: undefined: core.Genesis

Date: 2024-02-27 09:11:40-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: no field or method ChainID

Date: 2024-02-27 09:12:07-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: unresolve CommitGenesis

Date: 2024-02-27 09:13:20-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: fix logger defaults

Date: 2024-02-27 09:15:48-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: configs use slog Logger instead of glogger

Date: 2024-02-27 09:18:36-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: fixing up golang.org/x/expt/slog vs. log/slog vs. testlogger vs. log.Logger

Date: 2024-02-27 09:30:23-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: undefined: triedb

Date: 2024-02-27 09:32:04-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
internal/ethapi: undefined: core.Genesis

Date: 2024-02-27 09:32:56-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
common/math: var `u256_tt255` is unused; remove other unused vars too

Date: 2024-02-27 09:34:54-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: big -> uint256

Date: 2024-02-27 09:35:49-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
miner: undefined: taskIndex

Date: 2024-02-27 09:36:35-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core,internal/ethapi: cannot use types.Account{…} (value of type github.com/ethereum/go-ethereum/core/types.Account) as genesisT.GenesisAccount value in assignment

Date: 2024-02-27 10:16:59-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: assignment mismatch: 5 variables but test.RunNoVerifyWithPost returns 3 values

Date: 2024-02-27 10:18:09-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: cannot use func(err error, snaps *snapshot.Tree, state *state.StateDB) {…} (value of type func(err error, snaps *snapshot.Tree, state *state.StateDB)) as func(err error, st *StateTestState) value in argument to test.Run

Date: 2024-02-27 10:20:24-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core,miner: cannot use vars.EIP1234FBlockReward (variable of type *uint256.Int) as *big.Int value in argument to new(big.Int).Add

Date: 2024-02-27 10:38:42-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/ctypes,tests: config.IsCancun undefined

Date: 2024-02-27 11:31:30-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst,params/types/ctypes: adapt ForkChoiceUpdatedV2,3 methods to use configurator pattern (PTAL)

Date: 2024-02-27 12:31:59-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst,params/confp: fixup shanghai boolean, add Uint64Ptr2Big convenience function

Date: 2024-02-27 12:40:17-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
contracts/checkpointoracle: rm -rf; gone upstream too

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/tracers/internal/tracetest: assignment mismatch: 3 variables but tests.MakePreState returns 1 value

Date: 2024-02-27 12:42:17-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst,params/types/ctypes: remove empty logic branch

Date: 2024-02-27 12:48:31-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
internal/ethapi: file is not go-importsed

Date: 2024-02-27 12:49:36-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/checkpoint-admin: rm -rf

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
common/math: fix common.uint256 unmarshal behavior and tests

Holimant's uint256 lib is stricter than big
about its encoding. It demands leading 0[X|x]
and wont accept leading zeros.

Date: 2024-02-27 13:06:02-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params: remove custom ExtraData field for DeveloperGenesisBlock

I don't see any reason why this value
needs to get set.
Probably a test somewhere else relies
on it but I'll find that one later.

Date: 2024-02-28 06:20:57-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
ethclient: TestRPCDiscovre: remove OVER RPC method (method now gone)

Date: 2024-02-28 06:32:22-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
accounts/abi/bind: swap core.GenesisAlloc -> genesisT.GenesisAlloc in bind tests

Date: 2024-02-28 06:41:30-07:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
consensus/ethash,miner/stress/ethash: import correct slog package “golang.org/x/exp/slog”

meowsbits committed:
core: gaDeriveHash: only set balance if non-nil, avoid panic

Date: 2024-02-28 06:52:20-07:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
accounts/abi/bind: fix bind_test.go to use genesisT

meowsbits committed:
cmd/evm,core/vm: TestT8n fails because of a blobGasUsed mismatch

Date: 2024-02-28 10:13:13-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
tests: fix typo/wrongvar s/BlobTxGasPerBlob/MaxBlobGasPerBlob/

This is kind of a strange test situation
but at least its well documented.

This fixes a TestState failure, but
I'm still working on the TextT8n failure,
possibly related to some other blob dam
problems.

Date: 2024-02-28 15:14:42-07:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
cmd/evm/internal/t8ntool: cmd/evm/internal/t8ntool: fix merge typo for txBlobGas

ziogaschr committed:
consensus/ethash: complete verify the non-existence of cancun-specific header fields

cuishuang committed:
eth,eth/protocols/eth,internal/era,metrics,p2p/enode,rpc,signer/core: all: fix function names in docs (#29128)

Signed-off-by: cui fliter <imcusg@gmail.com>

Commit:
cmd/era,cmd/geth,cmd/utils,core/txpool,internal/era,internal/era/e2store,internal/era,miner,node,p2p/discover,p2p/discover/v5wire,p2p/dnsdisc,p2p/enode,p2p/nat,p2p/simulations/adapters,signer/core/apitypes,trie,triedb/pathdb: all: replace fmt.Errorf() with errors.New() if no param required (#29126)

replace-fmt-errorf

Co-authored-by: yzb@example.cn <yzb@example.cn>

meowsbits committed:
params/mutations: TestAccumlateRewards: fix typo occurred during merge

Date: 2024-02-29 05:58:26-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/forkid: TestGatherForks: install missing Cancun fork activation times

Date: 2024-02-29 06:01:21-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params: cannot use account.Balance (variable of type *big.Int) as *uint256.Int value in argument to s
tatedb.AddBalance

Date: 2024-02-29 06:03:24-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: verifyHeader: refactor EIP[4895,4788,4844] validations

Date: 2024-02-29 06:26:29-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/ctypes: tidy up w/r/t comments and documentation

Date: 2024-02-29 06:26:44-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: TestVerkleGenesisCommit: fix gaHash==hashAlloc genesis fn to respect verkle

Date: 2024-02-29 06:43:38-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/filters: TestSideBlockSubscription: fix panic b/c nil config on genesis

Date: 2024-02-29 07:30:31-07:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
miner: miner: worker_test.go:300: receipt number mismatch: have 0, want 1

meowsbits committed:
core: GenerateChain: condition blobGasPrice assignment on EIP4844 feature

Date: 2024-02-29 08:14:15-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
internal/ethapi: TestEstimateGas: fix invalid difficulty for wrapped engine (want min, got 0)

Date: 2024-02-29 08:14:56-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
internal/ethapi: TestCall: use MergedTestChainConfig (==upstream), SetPos only if config configs it

Date: 2024-02-29 09:07:15-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm,internal/ethapi: install new cg-specific TestCall with pre-merge config

Date: 2024-02-29 09:07:41-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/devp2p/internal/ethtest: TestEthSuite,TestSnapSuite: configure ethash ModeFake, fixes tests

The testdata use ethash as their consensus engine,
but define 0-values for all mixhashes.
So we need to not validate the PoW seals
for the test.

See the comment about how we might create a second
testdata suite that would actually define
mixhashes which could be validated by a real ethash
engine.

Date: 2024-02-29 09:40:40-07:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
tests: tests: fix TestDifficulty

--- FAIL: TestDifficulty (0.01s)
    --- FAIL: TestDifficulty/blockgenesistest.json (0.00s)
panic: json: cannot unmarshal array into Go value of type map[string]map[string]json.RawMessage in file testdata/BasicTests/blockgenesistest.json

ziogaschr committed:
accounts: fix HdPath tests to use SetCoinTypeConfiguration

meowsbits committed:
tests: update submodule tests/testdata to ::release/1.13 version

Date: 2024-03-01 14:21:37-07:00
Signed-off-by: meows <b5c6@protonmail.com>

yasyzb committed:
p2p: p2p: remove unused argument 'flags' (#29132)

meowsbits committed:
Merge branch 'release/1.13' into merge/foundation-release/1.13.14

Conflicts:
      core/txpool/blobpool/blobpool.go
      eth/catalyst/api.go
      eth/catalyst/api_test.go
      params/version.go

meowsbits committed:
core/txpool: s/params/vars/ BlobTxMinBlobGasprice

Date: 2024-03-04 09:46:29-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool/blobpool: undefined: params.BlobTxMinBlobGasprice

Date: 2024-03-04 09:48:08-07:00
Signed-off-by: meows <b5c6@protonmail.com>

cuiweixie committed:
rlp: rlp: using unsafe.Slice instead of SliceHeader (#29067)


Co-authored-by: Felix Lange <fjl@twurst.com>

meowsbits committed:
tests: execStateTest: actually skip forks that are registered to be skipped

Date: 2024-03-06 11:57:50-07:00
Signed-off-by: meows <b5c6@protonmail.com>

keroro520 committed:
p2p/simulations/adapters: p2p/simulations/adapters: fix error messages in TestTCPPipeBidirections (#29207)

fjl committed:
p2p: p2p: fix race in dialScheduler (#29235)


Co-authored-by: Stefan <stefan@starflinger.eu>

holiman committed:
.travis.yml,Dockerfile,Dockerfile.alltools,accounts/scwallet,build,crypto,crypto/ecies,crypto/secp256k1,crypto,go.sum,p2p/rlpx: all: update to go version 1.22.1 (#28946)

Since Go 1.22 has deprecated certain elliptic curve operations, this PR removes
references to the affected functions and replaces them with a custom implementation
in package crypto. This causes backwards-incompatible changes in some places.

---------

Co-authored-by: Marius van der Wijden <m.vanderwijden@live.de>
Co-authored-by: Felix Lange <fjl@twurst.com>

diega commented on .github/workflows/go.yml:
```diff
@@ -0,0 +1,23 @@
+
+jobs:
+  build:
+    runs-on: self-hosted
```
we would probably need to configure this runner on our side

diega commented on .travis.yml:
```diff
@@ -0,0 +1,168 @@
+language: go
+go_import_path: github.com/ethereum/go-ethereum
```
besides changing this, we could try to configure TravisCI to see if we can match some of this builds

diega commented on Dockerfile.alltools:
```diff
@@ -4,7 +4,7 @@ ARG VERSION=""
 
 # Build Geth in a stock Go builder container
-FROM golang:1.20-alpine as builder
+FROM golang:1.21-alpine as builder
```
Add to notes: bump min Go version to 1.21

ziogaschr replied:
`go.mod` still uses go1.20. Did this mismatch came from upstream?

ziogaschr replied:
Fixed by [b9eb08a](https://github.com/etclabscore/core-geth/pull/623/commits/b9eb08a87b113443110930fc3537832e7a6631c3) and [86f3184](https://github.com/etclabscore/core-geth/pull/623/commits/86f3184415f73e5db5b9e30767d900ce00ba8a13).

ziogaschr replied:
> Add to notes: bump min Go version to 1.21

I added it in this PR description so as to keep it for later

diega commented on cmd/devp2p/README.md:
```diff
@@ -108,31 +108,32 @@ Start the test by running `devp2p discv5 test -listen1 127.0.0.1 -listen2 127.0.
+        --verbosity 5                  \
+        --authrpc.jwtsecret 0x7365637265747365637265747365637265747365637265747365637265747365
+
+Note that the tests also require access to the engine API.
```
Based on this note I'm not sure if we should integrate `devp2p` changes without checking if it still works for our diagnosing usage

diega commented on cmd/devp2p/rlpxcmd.go:
```diff
@@ -46,22 +47,30 @@ var (
 		Name:      "eth-test",
-		Usage:     "Runs tests against a node",
-		ArgsUsage: "<node> <chain.rlp> <genesis.json>",
+		Usage:     "Runs eth protocol tests against a node",
```
AFAIK we're not using these, but worth to highlight that we'll not be able to use them ever now

ziogaschr replied:
Nice catch, as I used it once in past.
This haven't dropped support for them, just changed the "ArgsUsage".

diega commented on cmd/evm/runner.go:
```diff
@@ -146,11 +146,11 @@ func runCmd(ctx *cli.Context) error {
 		}
 	} else {
-		genesisConfig.Config = params.AllEthashProtocolChanges
+		genesisConfig.Config = params.AllDevChainProtocolChanges
```
would they be wiping out every _ethash_ concept by replacing them with _dev_?

ziogaschr replied:
They will actually continue supporting only post-merge chains

diega commented on cmd/geth/main.go:
LES finally deprecated

cuiweixie committed:
p2p/dnsdisc: p2p/dnsdisc: using maps.Copy (#29377)

weiihann committed:
p2p: p2p: add inbound and outbound peers metric (#29424)

cuiweixie committed:
p2p/dnsdisc: p2p/dnsdisc: using clear builtin func (#29418)


Co-authored-by: Felix Lange <fjl@twurst.com>

rjl493456442 committed:
eth/protocols/snap: skip retrieval for completed storages (#29378)

* eth/protocols/snap: skip retrieval for completed storages

* eth/protocols/snap: address comments from peter

* eth/protocols/snap: add comments

holiman committed:
core/rawdb: add sanity-limit to header accessor (#29534)

holiman committed:
eth/filters: enforce topic-limit early on filter criterias (#29535)

This PR adds a limit of 1000 to the "inner" topics in a filter-criteria

rjl493456442 committed:
core, eth/protocols/snap, trie: fix cause for snap-sync corruption, implement gentrie (#29313)

This pull request defines a gentrie for snap sync purpose.

The stackTrie is used to generate the merkle tree nodes upon receiving a state batch. Several additional options have been added into stackTrie to handle incomplete states (either missing states before or after).

In this pull request, these options have been relocated from stackTrie to genTrie, which serves as a wrapper for stackTrie specifically for snap sync purposes.

Further, the logic for managing incomplete state has been enhanced in this change. Originally, there are two cases handled:

-    boundary node filtering
-    internal (covered by extension node) node clearing

This changes adds one more:

- Clearing leftover nodes on the boundaries.

This feature is necessary if there are leftover trie nodes in database, otherwise node inconsistency may break the state healing.

ziogaschr commented on cmd/utils/flags.go:
```diff
@@ -1543,58 +1545,24 @@ func SetP2PConfig(ctx *cli.Context, cfg *p2p.Config) {
-
 	if ctx.IsSet(MaxPeersFlag.Name) {
 		cfg.MaxPeers = ctx.Int(MaxPeersFlag.Name)
-		if lightServer && !ctx.IsSet(LightMaxPeersFlag.Name) {
```
Once we release this version, we have to update our nodes to stop serving light clients.

ziogaschr commented on cmd/utils/flags.go:
```diff
@@ -2272,11 +2245,26 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 			if rawdb.ReadCanonicalHash(chaindb, 0) != (common.Hash{}) {
 				cfg.Genesis = nil // fallback to db content
+
+				// validate genesis has PoS enabled in block 0
```
We need to test `--dev` mode?

ziogaschr replied:
`--dev.pow` works. It mines blocks really fast, do you remember if it was waiting some time between blocks by default?

meowsbits replied:
It's too fast, like 1000 blocks / 10 seconds. Its probably using `--fakepow` with no pause. I'll take a look at it.

meowsbits replied:
Actually... I only get the super-fast blocks when I use `./build/bin/geth --dev.pow --fakepow.poisson`. If I just run it simply with `./build/bin/geth --dev.pow`, after startup geth waits for me to `geth attach /tmp/geth.ipc` and run `miner.start()`, and then it produces blocks at a reasonable pace. So I think if anything's broken here, it's probably the `--fakepow.poisson` which wasn't exactly designed to be used with developer mode anyways.

I think there's nothing to fix.

ziogaschr commented on consensus/ethash/ethash_test.go:
```diff
@@ -27,16 +27,15 @@ import (
-	glogger := log.NewGlogHandler(log.StreamHandler(os.Stdout, log.TerminalFormat(false)))
-	glogger.Verbosity(log.Lvl(99))
-	log.Root().SetHandler(glogger)
+	slog.SetDefault(slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{Level: slog.LevelDebug})))
```
Why we don't use the `go-ethereum/log` here?

ziogaschr replied:
"golang.org/x/exp/slog" is used in the log package either way. But can't we use that instead and keep the dependency just there?

ziogaschr commented on core/genesis.go:
```diff
@@ -98,15 +100,9 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, triedb *trie.Database, gen
-			}
 			if overrides != nil && overrides.OverrideCancun != nil {
 				config.SetEIP4844TransitionTime(overrides.OverrideCancun)
+				// TODO(meowsbits) Install the remaining Cancun EIP overrides.
```
Note to not forget it

ziogaschr replied:
Fixed with [9290ad9](https://github.com/etclabscore/core-geth/pull/623/commits/9290ad92a48c0a810bfa8916da8646e4cd77a462)

holiman committed:
params: release Geth v 1.13.15

ziogaschr commented on params/types/ctypes/configurator_iface.go:
```diff
@@ -148,58 +149,73 @@ type ProtocolSpecifier interface {
-	// ModExp gas cost
+	// Berlin:
+
+	// GetEIP2565Transition implements EIP-2565: ModExp Gas Cost - https://eips.ethereum.org/EIPS/eip-2565
```
@meowsbits love you for the comments :)

ziogaschr commented on cmd/faucet/faucet.go:
```diff
@@ -1,1244 +0,0 @@
-// Copyright 2017 The go-ethereum Authors
```
@meowsbits do we want to keep and maintain faucet as you were using it?

meowsbits replied:
I was using it for Mordor. If we have, or there are, other ways of distributing tokens then I'm fine abandoning or archiving it.

ziogaschr replied:
@diega I think you also have a faucet. Which backend does it use?

diega replied:
It's just a CF worker that gives away 1 mETC to any address with less than 2 mETC that I fill from the mining rewards of the Besu node that I maintain

ziogaschr commented on eth/downloader/downloader_test.go:
```diff
@@ -752,9 +723,6 @@ func testMultiSynchronisation(t *testing.T, protocol uint, mode SyncMode) {
 func TestMultiProtoSynchronisation68Full(t *testing.T)  { testMultiProtoSync(t, eth.ETH68, FullSync) }
 func TestMultiProtoSynchronisation68Snap(t *testing.T)  { testMultiProtoSync(t, eth.ETH68, SnapSync) }
 func TestMultiProtoSynchronisation68Light(t *testing.T) { testMultiProtoSync(t, eth.ETH68, LightSync) }
-func TestMultiProtoSynchronisation67Full(t *testing.T)  { testMultiProtoSync(t, eth.ETH67, FullSync) }
```
Do we want to keep ETH67 and its tests?

ziogaschr replied:
What is the latest version implemented by besu and Erigon?

ziogaschr replied:
This is the PR removing it https://github.com/ethereum/go-ethereum/pull/28956

ziogaschr replied:
PR that reverts the ETH67 removal https://github.com/etclabscore/core-geth/pull/629

ziogaschr replied:
Ohh, @meowsbits @diega we have to merge this one I think. If yes, then we have to rerun our sync test.

ziogaschr committed:
Merge branch 'foundation-release/1.13.15' into merge/foundation-release/1.13.15

# Conflicts:
#	params/version.go

ziogaschr committed:
go.mod,go.sum: go mod tidy

ziogaschr committed:
go.mod: set go.mod version to 1.21

ziogaschr committed:
go.mod,go.sum: go mod tidy

ziogaschr committed:
.github/workflows: update GitHub workflows to use go version 1.21

ziogaschr committed:
.github/workflows,go.mod,go.sum: Merge branch 'merge/foundation-release/1.13.14' into merge/foundation-release/1.13.15

ziogaschr committed:
core: install the remaining Cancun EIP overrides

ziogaschr commented on eth/protocols/eth/handler.go:
We can probably remove this comment https://github.com/etclabscore/core-geth/pull/623/files#diff-5beb4052baf94e3025566d929c63974350e4b1eae18cce8eb21b4579d6788ffeR98.
@meowsbits is there anything else to do about it?

meowsbits replied:
My browser is struggling to load the linked comment. Please clarify which comment you're pointing to.

ziogaschr replied:
My browser can't load it either. It was probably a comment in code with a TODO on your name.

kiarash8112 committed:
p2p/discover/v5wire: p2p/discover/v5wire: add tests for invalid handshake and auth data size (#29708)

ziogaschr committed:
params/vars: params/vars: update SupportedProtocolVersions

fjl committed:
cmd/devp2p,internal/testlog,node,p2p/discover,p2p: p2p/discover: improved node revalidation (#29572)

Node discovery periodically revalidates the nodes in its table by sending PING, checking
if they are still alive. I recently noticed some issues with the implementation of this
process, which can cause strange results such as nodes dropping unexpectedly, certain
nodes not getting revalidated often enough, and bad results being returned to incoming
FINDNODE queries.

In this change, the revalidation process is improved with the following logic:

- We maintain two 'revalidation lists' containing the table nodes, named 'fast' and 'slow'.
- The process chooses random nodes from each list on a randomized interval, the interval being
  faster for the 'fast' list, and performs revalidation for the chosen node.
- Whenever a node is newly inserted into the table, it goes into the 'fast' list.
  Once validation passes, it transfers to the 'slow' list. If a request fails, or the
  node changes endpoint, it transfers back into 'fast'.
- livenessChecks is incremented by one for successful checks. Unlike the old implementation,
  we will not drop the node on the first failing check. We instead quickly decay the
  livenessChecks give it another chance.
- Order of nodes in bucket doesn't matter anymore.

I am also adding a debug API endpoint to dump the node table content.

Co-authored-by: Martin HS <martin@swende.se>

fjl committed:
p2p/enode,p2p/enr: p2p/enode: fix endpoint determination for IPv6 (#29801)

enode.Node has separate accessor functions for getting the IP, UDP port and TCP port.
These methods performed separate checks for attributes set in the ENR.

With this PR, the accessor methods will now return cached information, and the endpoint is
determined when the node is created. The logic to determine the preferred endpoint is now
more correct, and considers how 'global' each address is when both IPv4 and IPv6 addresses
are present in the ENR.

AaronChen0 committed:
p2p/enode: p2p/enode: fix TCPEndpoint (#29827)

gitglorythegreat committed:
p2p/discover,p2p: p2p: fix typos (#29828)

fjl committed:
p2p/discover: p2p/discover: fix crash when revalidated node is removed (#29864)

In #29572, I assumed the revalidation list that the node is contained in could only ever
be changed by the outcome of a revalidation request. But turns out that's not true: if the
node gets removed due to FINDNODE failure, it will also be removed from the list it is in.
This causes a crash.

The invariant is: while node is in table, it is always in exactly one of the two lists. So
it seems best to store a pointer to the current list within the node itself.

lightclient committed:
p2p/discover: p2p/discover: fix update logic in handleAddNode (#29836)

It seems the semantic differences between addFoundNode and addInboundNode were lost in
#29572. My understanding is addFoundNode is for a node you have not contacted directly
(and are unsure if is available) whereas addInboundNode is for adding nodes that have
contacted the local node and we can verify they are active.

handleAddNode seems to be the consolidation of those two methods, yet it bumps the node in
the bucket (updating it's IP addr) even if the node was not an inbound. This PR fixes
this. It wasn't originally caught in tests like TestTable_addSeenNode because the
manipulation of the node object actually modified the node value used by the test.

New logic is added to reject non-inbound updates unless the sequence number of the
(signed) ENR increases. Inbound updates, which are published by the updated node itself,
are always accepted. If an inbound update changes the endpoint, the node will be
revalidated on an expedited schedule.

Co-authored-by: Felix Lange <fjl@twurst.com>

bugmaker9371 committed:
p2p/simulations/adapters,p2p/simulations/examples: p2p/simulations: remove stale information about docker adapter (#29874)

lilasxie committed:
: p2p/nodestate: remove unused package (#29872)

fjl committed:
cmd/devp2p/internal/v4test,p2p/discover,p2p/discover/v4wire,p2p/discover,p2p: p2p/discover: refactor node and endpoint representation (#29844)

Here we clean up internal uses of type discover.node, converting most code to use
enode.Node instead. The discover.node type used to be the canonical representation of
network hosts before ENR was introduced. Most code worked with *node to avoid conversions
when interacting with Table methods. Since *node also contains internal state of Table and
is a mutable type, using *node outside of Table code is prone to data races. It's also
cleaner not having to wrap/unwrap *enode.Node all the time.

discover.node has been renamed to tableNode to clarify its purpose.

While here, we also change most uses of net.UDPAddr into netip.AddrPort. While this is
technically a separate refactoring from the *node -> *enode.Node change, it is more
convenient because *enode.Node handles IP addresses as netip.Addr. The switch to package
netip in discovery would've happened very soon anyway.

The change to netip.AddrPort stops at certain interface points. For example, since package
p2p/netutil has not been converted to use netip.Addr yet, we still have to convert to
net.IP/net.UDPAddr in a few places.

meowsbits committed:
Merge pull request #627 from etclabscore/merge/foundation-release/1.13.15

Merge/foundation release/1.13.15

fjl committed:
cmd/devp2p/internal/ethtest,cmd/devp2p/internal/v4test,cmd/devp2p,p2p,p2p/discover,p2p/discover/v5wire,p2p/enode,p2p/netutil,p2p: p2p: use netip.Addr where possible (#29891)

enode.Node was recently changed to store a cache of endpoint information. The IP address in the cache is a netip.Addr. I chose that type over net.IP because it is just better. netip.Addr is meant to be used as a value type. Copying it does not allocate, it can be compared with ==, and can be used as a map key.

This PR changes most uses of Node.IP() into Node.IPAddr(), which returns the cached value directly without allocating.
While there are still some public APIs left where net.IP is used, I have converted all code used internally by p2p/discover to the new types. So this does change some public Go API, but hopefully not APIs any external code actually uses.

There weren't supposed to be any semantic differences resulting from this refactoring, however it does introduce one: In package p2p/netutil we treated the 0.0.0.0/8 network (addresses 0.x.y.z) as LAN, but netip.Addr.IsPrivate() doesn't. The treatment of this particular IP address range is controversial, with some software supporting it and others not. IANA lists it as special-purpose and invalid as a destination for a long time, so I don't know why I put it into the LAN list. It has now been marked as special in p2p/netutil as well.

ziogaschr committed:
go.sum: go mod tidy

ziogaschr committed:
ethclient: fix: add `debug_discoveryV4Table` method for `TestRPCDiscover`

ziogaschr committed:
consensus/ethash: fix lint SA1019 reflect.SliceHeader

fjl committed:
p2p/discover: p2p/discover: unwrap 4-in-6 UDP source addresses (#29944)

Fixes an issue where discovery responses were not recognized.

ziogaschr committed:
consensus/ethash: more fixes

meowsbits committed:
Merge pull request #637 from etclabscore/merge/foundation-release/1.13.15-p2p-lint

Fix lint issue SA1019 reflect.SliceHeader

meowsbits committed:
Merge pull request #636 from etclabscore/merge/foundation-release/1.13.15-p2p

Add latest p2p

diega committed:
Merge pull request #639 from etclabscore/merge/foundation-release/1.13.15-p2p-discover-unwrap-4in6

Include latest foundation release fixes in p2p
